src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseParser.cs

```diff
@@ -223,21 +223,37 @@ private static unsafe int GetResponseHeader(SafeWinHttpHandle requestHandle, uin
         /// <summary>
         /// Returns the size of the char array buffer.
         /// </summary>
-        private static unsafe int GetResponseHeaderCharBufferLength(SafeWinHttpHandle requestHandle, uint infoLevel)
+        public static unsafe int GetResponseHeaderCharBufferLength(SafeWinHttpHandle requestHandle, bool isTrailingHeaders)
         {
             char* buffer = null;
             int bufferLength = 0;
             uint index = 0;
 
+            uint infoLevel = Interop.WinHttp.WINHTTP_QUERY_RAW_HEADERS_CRLF;
+            if (isTrailingHeaders)
+            {
+                infoLevel |= Interop.WinHttp.WINHTTP_QUERY_FLAG_TRAILERS;
+            }
+
             if (!QueryHeaders(requestHandle, infoLevel, buffer, ref bufferLength, ref index))
             {
                 int lastError = Marshal.GetLastWin32Error();
 
-                Debug.Assert(lastError != Interop.WinHttp.ERROR_WINHTTP_HEADER_NOT_FOUND);
+                if (!isTrailingHeaders)
+                {
+                    Debug.Assert(lastError != Interop.WinHttp.ERROR_WINHTTP_HEADER_NOT_FOUND);
 
-                if (lastError != Interop.WinHttp.ERROR_INSUFFICIENT_BUFFER)
+                    if (lastError != Interop.WinHttp.ERROR_INSUFFICIENT_BUFFER)
+                    {
+                        throw WinHttpException.CreateExceptionUsingError(lastError, nameof(Interop.WinHttp.WinHttpQueryHeaders));
+                    }
+                }
+                else
                 {
-                    throw WinHttpException.CreateExceptionUsingError(lastError, nameof(Interop.WinHttp.WinHttpQueryHeaders));
+                    if (!(lastError == Interop.WinHttp.ERROR_INSUFFICIENT_BUFFER || lastError == Interop.WinHttp.ERROR_WINHTTP_HEADER_NOT_FOUND))
+                    {
+                        throw WinHttpException.CreateExceptionUsingError(lastError, nameof(Interop.WinHttp.WinHttpQueryHeaders));
+                    }
```

Comment:
Seems like this would be simpler as:
```C#
if (lastError != Interop.WinHttp.ERROR_INSUFFICIENT_BUFFER &&
    (!isTrailingHeaders ||
     lastError != Interop.WinHttp.ERROR_WINHTTP_HEADER_NOT_FOUND))
{
    throw WinHttpException.CreateExceptionUsingError(lastError, nameof(Interop.WinHttp.WinHttpQueryHeaders));
}
```
or something like that?

Reply:
It also has to be proceeded by:
```C#
Debug.Assert(isTrailingHeaders || lastError != Interop.WinHttp.ERROR_WINHTTP_HEADER_NOT_FOUND);
```

Can't decide if the code more readable in this form, pushed a change anyways.